renku/core/template/usecase.py

```diff
@@ -132,7 +132,15 @@ def update_template(
             "break your project"
         )
 
-    templates_source = fetch_templates_source(source=template_metadata.source, reference=template_metadata.reference)
+    try:
+        templates_source = fetch_templates_source(
+            source=template_metadata.source, reference=template_metadata.reference
+        )
+    except errors.TemplateMissingReferenceError as e:
+        message = f"{str(e)}; You can still manually update the template and set a difference reference."
+        raise errors.TemplateUpdateError(message)
+    except errors.InvalidTemplateError:
+        raise errors.TemplateUpdateError("Template cannot be fetched.")
```

Comment:
What's the reason to convert these errors to `TemplateUpdateError`? I believe we don't need to catch them here.

Reply:
The idea was to simplify catching/handling them in the service and in the CLI. For the latter, throwing a new error results in a clean error message, while not doing it means a long and confusing stack trace is printed (see screen below).
Is there a better way to handle this? Or maybe that should be done in the cli code?
![Screenshot_20220413_134606](https://user-images.githubusercontent.com/43481553/163173260-789cf951-97b3-4248-9eb7-3d72e6779259.png)